### DISCUSSION TITLE
Consistent error when executing query to MySQL db

### DISCUSSION THREAD

**Discussion by danishkyas33n** [prisma/prisma#13057] [Q&A] [answered]
Receiving "Cannot read value of undefined ('create')" consistently every time I run a "Prisma.c_session.create()" query. Happens for all types of queries.

**Comment by nurul3101** [comment, marked as answer]
Hey @danishkyas33n 👋 

Have you generated your PrismaClient by executing `npx prisma generate`? 
Can you also share how are you initialising your PrismaClient?

**Reply by danishkyas33n** [reply]
Hello @nurul3101, yes, I have run *npx Prisma generate* and I have used the basic initialization provided in the docs;
.......................................................................

import { PrismaClient } from "@prisma/client";

const Prisma = new PrismaClient();

.......................................................................

I have also tried using a global instance like this;
.......................................................................

import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }

  prisma = global.prisma;
}

export default prisma;
.......................................................................

But I'm still getting the same error;
.......................................................................
TypeError: Cannot read properties of undefined (reading 'create')
    at addContractHandler (webpack-internal:///(api)/./pages/api/admin/contracts/addContract.js:130:54)
    at async Object.apiResolver (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\api-utils\node.js:182:9)
    at async DevServer.runApi (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\next-server.js:386:9)
    at async Object.fn (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\base-server.js:488:37)
    at async Router.execute (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\router.js:228:32)
    at async DevServer.run (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\base-server.js:600:29)
    at async DevServer.run (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\dev\next-dev-server.js:445:20)
    at async DevServer.handleRequest (F:\DankTech\SACOF\sacof-app\node_modules\next\dist\server\base-server.js:307:20)
.......................................................................

**Reply by danishkyas33n** [reply]
Some assistance on this matter would be greatly appreciated as it's a time sensitive project. Thanks in advance.

**Reply by janpio** [reply]
It looks like your Client did not actually get created. I suggest you start a new project, add Prisma and Prisma Client, modify the schema to have 1 model, and then see if you can use that model's API in your script.

**Reply by danishkyas33n** [reply]
Guys, I figured out the issue. The error was occuring because of the naming conventions.

So I created a model;
.......................................................................


model shippingInstruction {
  id                    String @id @default(cuid())
  image_name String @db.VarChar(255)

  @@map(name: "shippinginstructions")
}

.......................................................................

And my query;
.......................................................................

await prisma.shippinginstructions.create({
        data: {
          image_name: "test",
        }
      });

.......................................................................

I changed the model name to just "shipping" and everything is working fine.

But this is not my desired model name, would be better to use the name I need rather than one that suits Prisma.

Please look into this. 🥺🙏🏽
Thanks in advance.

**Reply by janpio** [reply]
If you call the model `shippingInstruction`, then the Javascript call needs to be `prisma.shippingInstruction.create()`. 
(Only the first letter is lowercased for the Prisma Client API, which does not affect you)

**Reply by danishkyas33n** [reply]
Ohhhh my days, I've been using the "@@map("shippinginstructions") name for queries! 🙆🏽🤦🏽‍♂️

Sorry guys my bad, keep up the good work!

Love Prisma tho, really fun to use!

*Lesson learnt*
🥺

**Reply by janpio** [reply]
You are not alone in that confusion. I wonder if we could do anything to surface that better.

**Reply by danishkyas33n** [reply]
I guess a small paragraph on the docs mentioning that all calls will be made to the model name and not the mapped name.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
